OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs

```diff
@@ -64,6 +64,50 @@ public FrozenUnderFog(ActorInitializer init, FrozenUnderFogInfo info)
 			footprint = footprintCells.SelectMany(c => map.ProjectedCellsCovering(c.ToMPos(map))).ToArray();
 		}
 
+		public void Created(Actor self)
+		{
+			tooltip = self.TraitsImplementing<ITooltip>().FirstOrDefault();
+			health = self.TraitOrDefault<Health>();
+		}
+
+		void UpdateFrozenActor(Actor self, FrozenActor frozenActor, int playerIndex)
+		{
+			VisibilityHash |= 1 << (playerIndex % 32);
+
+			frozenActor.Owner = self.Owner;
+
+			if (health != null)
+			{
+				frozenActor.HP = health.HP;
+				frozenActor.DamageState = health.DamageState;
+			}
+
+			if (tooltip != null)
+			{
+				frozenActor.TooltipInfo = tooltip.TooltipInfo;
+				frozenActor.TooltipOwner = tooltip.Owner;
+			}
+		}
+
+		public void AddedToWorld(Actor self)
+		{
+			if (initialized)
+				return;
```

Comment:
Will using `INotifyCreated` instead work for this? Would be nice to avoid having to have this flag.